extensions/wikia/WikiAnswers/templates/WikiAnswers_QuestionBox.php

```diff
@@ -11,7 +11,22 @@ function askQuestion(){
 	q = encodeURIComponent( q );
 
 	var path = window.wgServer + window.wgArticlePath.replace("$1","");
-	window.location = path + "Special:CreateQuestionPage?questiontitle=" + q.charAt(0).toUpperCase() + q.substring(1);
+
+	jQuery.ajax({
+		url: path + "Special:CreateQuestionPage",
+		type: 'POST',
+		data: {
+			'questiontitle': q.charAt(0).toUpperCase() + q.substring(1),
+			'token': window.mw.user.tokens.get('editToken')
+		}
+	}).done(function () {
+		window.location.href = path + "Special:CreateQuestionPage?questiontitle="
+			+ q.charAt(0).toUpperCase() + q.substring(1);
+	}).fail(function () {
+		console.log("Error creating question!");
```

Comment:
@kvas-damian: The controller code is not designed for XHR requests, so there isn't an easy way to robustly return a suitable error message in response. I did discuss this with @garthwebb, and we decided it is not worth the refactoring at the moment to display the error to the user as the ROI on this is very low given the negligible usage of this extension.
I did add a log to be able to verify user's process was in fact processed if a question was raised.